src/Optimizer/Transformer/ServerSideRendering.php

```diff
@@ -526,12 +484,8 @@ private function applyLayoutAttributes(DOMElement $element, $layout, CssLength $
                 break;
         }
 
-        // We prepend just in case an existing value (which shouldn't be there for valid docs) doesn't end with ';'.
-        if ($element->hasAttribute(Tag::STYLE)) {
-            $styles .= $element->getAttribute(Tag::STYLE);
-        }
-        if (! empty($styles)) {
-            $element->setAttribute(Tag::STYLE, $styles);
+        if (!empty($styles)) {
+            $element->addInlineStyle($styles);
```

Comment:
This could also throw an exception, and when that happens, it should get caught in `\AmpProject\Optimizer\Transformer\ServerSideRendering::applyLayout()` and then added to to `$errors` to report that SSR failed to complete, right?

Reply:
I safeguarded both this one and the next by wrapping at a higher level and creating a clone first so we can leave the element in an uncorrupted state:
```php
        try {
            $newElement = $element->cloneNode(true);
            $this->applyLayoutAttributes($element, $layout, $width, $height);
            $this->maybeAddSizerInto($document, $element, $layout, $width, $height);
            $element->parentNode->replaceChild($newElement, $element);
        } catch (MaxCssByteCountExceeded $exception) {
            $errors->add(
                Error\CannotPerformServerSideRendering::fromMaxCssByteCountExceededException($exception, $element)
            );
            return false;
        }
```

That seems to be the cleanest solution. Anything else might catch the exceptions, but will leave half-transformed left-overs in the DOM.